web/src/elements/ak-checkbox-group/ak-checkbox-group.ts

```diff
@@ -0,0 +1,212 @@
+
+type CheckboxKv = { name: string; label: string | TemplateResult };
+type CheckboxPr = [string, string | TemplateResult];
+export type CheckboxPair = CheckboxKv | CheckboxPr;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const isCheckboxPr = (t: any): t is CheckboxPr => Array.isArray(t);
+function* kvToPairs(items: CheckboxPair[]): Iterable<CheckboxPr> {
+    for (const item of items) {
+        yield isCheckboxPr(item) ? item : [item.name, item.label];
+    }
+}
+
+const AkElementWithCustomEvents = CustomEmitterElement(AKElement);
+
+/**
+ * @element ak-checkbox-group
+ *
+ * @class CheckboxGroup
+ *
+ * @description
+ * CheckboxGroup renders a collection of checkboxes in a linear list. Multiple
+ * checkboxes may be picked.
+ *
+ * @attr {options} - An array of either `[string, string | TemplateResult]` or
+ *     `{ name: string, label: string | TemplateResult }`. The first value or
+ *     `name` field must be a valid HTML identifier compatible with the HTML
+ *     `name` attribute.
+ *
+ * @attr {value} - An array of `name` values corresponding to the options that
+ *     are selected when the element is rendered.
+ *
+ * @attr {name} - The name of this element as it will appear in any <form>
+ *     transaction
+ *
+ * @attr {required} - If true, and if name is set, and no values are chosen,
+ *     will automatically fail a form `submit` event, providing a warning
+ *     message for any labeling. Note: if `name` is not set, this has no effect,
+ *     and a warn() will appear on the console.
+ *
+ * @event {input} - Fired when the component's value has changed. Current value
+ *     as an array of `name` will be in the `Event.detail` field.
+ *
+ * @event {change} - Fired when the component's value has changed. Current value
+ *     as an array of `name` will be in the `Event.detail` field.
+ *
+ * @csspart checkbox - The div containing the checkbox item and the label
+ * @csspart label - the label
+ * @csspart input - the input item
+ * @csspart checkbox-group - the wrapper div with flexbox control
+ *
+ * ## Bigger hit area
+ *
+ * Providing properly formatted names for selections allows the element to
+ * associate the label with the event, so the entire horizontal area from
+ * checkbox to end-of-label will be the hit area.
+ *
+ * ## FormAssociated compliance
+ *
+ * If a <form> component is a parent, this component will correctly send its
+ * values to the form for `x-form-encoded` data; multiples will appear in the
+ * form of `name=value1&name=value2` format, and must be unpacked into an array
+ * correctly on the server side according to the CGI (common gateway interface)
+ * protocol.
+ *
+ */
+
+@customElement("ak-checkbox-group")
+export class CheckboxGroup extends AkElementWithCustomEvents {
+    static get styles() {
+        return [
+            PFBase,
+            PFForm,
+            PFCheck,
+            css`
+                .pf-c-form__group-control {
```

Comment:
Taken from https://github.com/goauthentik/authentik/blob/main/web/src/elements/forms/Radio.ts#L40, idk if we want to dedupe that @kensternberg-authentik

Reply:
Not yet, I don't think.  Good catch.  Thanks so much!